File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/channel/metadata/ChannelMetadataId.java
package com.example.chat.domain.channel.metadata;

import java.util.UUID;

/**
 * 채팅방 메타데이터 ID (Value Object)
 */
public record ChannelMetadataId(String value) {
    public ChannelMetadataId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("ChannelMetadataId cannot be blank");
        }
    }

    public static ChannelMetadataId of(String value) {
        return new ChannelMetadataId(value);
    }

    public static ChannelMetadataId generate() {
        return new ChannelMetadataId(UUID.randomUUID().toString());
    }

    @Override
    public String toString() {
        return value;
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/channel/metadata/ChannelMetadataId.kt
package com.example.chat.domain.channel.metadata

import java.util.*

/**
 * 채팅방 메타데이터 ID (Value Object)
 */
@JvmInline
value class ChannelMetadataId(val value: String) {
	init {
		require(value.isNotBlank()) { "ChannelMetadataId cannot be blank" }
	}

	companion object {
		fun of(value: String) = ChannelMetadataId(value)
		fun generate() = ChannelMetadataId(UUID.randomUUID().toString())
	}

	override fun toString(): String = value
}
